### PR TITLE
split long load method in command

### PR DESCRIPTION
**Pull request by tomas-pecserke**
Prep work for extracting functionality from command into separate services. Extracted tasks into separate functions.

### CODE REVIEW

**Comment by tomas-pecserke** [comment]
@mhlavac Could you do code review please?

**Comment by tomas-pecserke** [comment]
@mhlavac I'd like to decouple logic from console interface using events and register listeners from the command once it starts. It's the only place I have Input and Output. What do you think?

File: Command/LoadYamlFixturesCommand.php [hunk 1]
```diff
@@ -101,30 +124,54 @@ protected function execute(InputInterface $input, OutputInterface $output)
             }
         }
 
-        $parser = new YamlFixtureFileParser();
-        $fixturesData = $parser->parse($fixtureFiles);
-        if (empty($fixturesData)) {
-            $output->writeln('  <info>No fixtures to load</info>');
+        return$fixtureFiles;
```
Comment (mhlavac) [inline Command/LoadYamlFixturesCommand.php]: missing space between `return` and `$fixtureFiles`

File: Command/LoadYamlFixturesCommand.php [hunk 2]
```diff
@@ -76,10 +77,32 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+
+        $this->load($fixturesData, $orm, $odm, $output);
+    }
+
+    /**
+     * @param Kernel $kernel
+     * @param bool $dirOrFile
+     * @return string[]
+     */
+    private function locateFixtures(Kernel $kernel, $dirOrFile) {
```
Comment (mhlavac) [inline Command/LoadYamlFixturesCommand.php]: small things... `{` should be on new line

File: Command/LoadYamlFixturesCommand.php [hunk 3]
```diff
@@ -101,30 +124,54 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        return$fixtureFiles;
+    }
 
-            return;
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return bool
+     */
+    private function confirmPurge(InputInterface $input, OutputInterface $output) {
```
Comment (mhlavac) [inline Command/LoadYamlFixturesCommand.php]: `{` should be on new line

File: Command/LoadYamlFixturesCommand.php [hunk 4]
```diff
@@ -101,30 +124,54 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-            foreach (($orm !== null ? $orm->getManagers() : array()) as $em) {
-                $ormPurger->setEntityManager($em);
-                $ormPurger->purge();
-            }
+    /**
+     * @param RegistryInterface $orm
+     * @param RegistryInterface $odm
+     * @param bool $truncate
+     */
+    private function purge(RegistryInterface $orm, RegistryInterface $odm, $truncate) {
```
Comment (mhlavac) [inline Command/LoadYamlFixturesCommand.php]: `{` should be on new line

**Comment by mhlavac** [comment]
@tomas-pecserke looks better. Doing events is fine, it means anyone can do something before and after fixtures are loaded.

Another option is to collect necessary values from input/output, ask questions and then call a service with all gathered options.

I think Command should be lightweight same as Controller. It can just parse request data and sends it to another service or services.

**Comment by tomas-pecserke** [comment]
@mhlavac That's the idea. I want the command to be interactive and report progress. Using the events I can still track the progress.

**Pull request merged**